From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NONPLAYT <76615486+NONPLAYT@users.noreply.github.com>
Date: Sun, 6 Jul 2025 03:04:04 +0300
Subject: [PATCH] Carpet-Fixes: Fix Slime Block Bounce Logic

This patch is based on the following mixins:
* "carpetfixes/mixins/blockFixes/SlimeBlock_incorrectLogicMixin.java"
By: fxmorin <28154542+fxmorin@users.noreply.github.com>
As part of: carpet-fixes (https://github.com/fxmorin/carpet-fixes)
Licensed under: MIT (https://opensource.org/licenses/MIT)

Patch description:
Fixes incorrect logic in the slimeblock bounce code which prevents some entities from bouncing.
This bug in the slime code is due to onGround reversing the velocity, the best way to fix this issue is by
setting onGround to true, only once the Y velocity is smaller than -0.15

diff --git a/net/minecraft/world/level/block/SlimeBlock.java b/net/minecraft/world/level/block/SlimeBlock.java
index f26c3100d0aad0227f016bfa4c86ac647e13485a..59c9e63e6d5634973cc4927eac479463cc486a4a 100644
--- a/net/minecraft/world/level/block/SlimeBlock.java
+++ b/net/minecraft/world/level/block/SlimeBlock.java
@@ -42,6 +42,7 @@ public class SlimeBlock extends HalfTransparentBlock {
         Vec3 deltaMovement = entity.getDeltaMovement();
         if (deltaMovement.y < 0.0) {
             double d = entity instanceof LivingEntity ? 1.0 : 0.8;
+            if (org.bxteam.divinemc.config.DivineConfig.FixesCategory.fixIncorrectBounceLogic) entity.setOnGround(deltaMovement.y > -0.15); // DivineMC - Carpet-Fixes: Fix Slime Block Bounce Logic
             entity.setDeltaMovement(deltaMovement.x, -deltaMovement.y * d, deltaMovement.z);
         }
     }
